scripts/webhook-migration/updateWebhooks.sh

```diff
@@ -0,0 +1,223 @@
+getGitHubTokens() {
+  for i in "${!OSIO_TOKENS[@]}"; do
+    [ -z "${OSIO_TOKENS[${i}]}"  ] && echo "WARN: Empty OSIO token for namespace ${NAME_SPACES[$i]}" >&2 && continue
+
+    gh_token=$(curl -sgSL -H "Authorization: Bearer ${OSIO_TOKENS[${i}]}" ${AUTH_URL}api/token?for=https://github.com | jq -r '.access_token')
+    if [ "${gh_token}" == "null" ]; then
+        echo "WARN: Unable to create GitHub token for UUID ${UUIDS[$i]}" >&2
+        curl -sgSL -H "Authorization: Bearer ${OSIO_TOKENS[${i}]}" ${AUTH_URL}api/token?for=https://github.com >&2
+    else
+        GITHUB_TOKENS[$i]=${gh_token}
+    fi
+  done
+}
+
+###############################################################################
+# For each OSIO token get the GitHub token
+# See https://developer.github.com/v3/repos/hooks
+# Globals:
+#   REPOS          - List of repositories to update
+#   GITHUB_TOKENS  - For each repository its GitHub access token
+#   NAME_SPACES    - For each repository its namespace (for logging)
+# Arguments:
+#   None
+# Returns:
+#   None
+###############################################################################
+updateGitHubWebhooks() {
+  for i in "${!REPOS[@]}"; do
+    [ -z "${GITHUB_TOKENS[${i}]}"  ] && echo "WARN: Empty GitHub token for namespace ${NAME_SPACES[$i]}" >&2 && continue
+
+    while read id url; do
+        if [[ ${url} =~ .*${JENKINS_OLD_URL_SUFFIX}.*$ ]]; then
+          if [ "${DRY_RUN}" = true ] ; then
+            echo DRY_RUN for ${url} update:
+            echo curl -sgSL -H "Authorization: token ***" -X PATCH \
+            --data '{"config": {"url": "'${JENKINS_PROXY_URL}'github-webhook/"}}' https://api.github.com/repos/${REPOS[$i]}/hooks/${id}
+            echo " "
+          else
+            response=$(curl -sgSL -H "Authorization: token ${GITHUB_TOKENS[$i]}" -X PATCH \
+              --data '{"config": {"url": "'${JENKINS_PROXY_URL}'github-webhook/"}}' \
+              https://api.github.com/repos/${REPOS[$i]}/hooks/${id} | \
+              jq -r '.last_response.message')
+            echo "${url} updated. Status: $response"
+          fi
+        else
+          echo "Skipping GitHub webhook '${url}' (no match)"
+        fi
+    done < <(curl -sgSL -H "Authorization: token ${GITHUB_TOKENS[$i]}" \
+      https://api.github.com/repos/${REPOS[$i]}/hooks |  jq -r '.[] |  if .id? != null then (.id|tostring) + " " + .config.url else "- -"  end')
+  done
+}
+
+###############################################################################
+# Checks whether the specified string ends with a '/' and if not adds one.
+# Globals:
+#   None
+# Arguments:
+#   $1       - The string to check for the trailing slash.
+# Returns:
+#   The unmodified string as passed to the function in case the string already has
+#   a trailing slash, otherwise the passed string with a '/' appended.
+###############################################################################
+ensureTrailingSlash() {
```

Comment:
I'd probably do
```
ensureTrailingSlash() {
  echo $1 | sed 's#\([^/]\)#\1/#'
}
```
:)

Reply:
LOL, not sure that is so much easier ;-) TBH, I usually reach for `sed` for these cases as well, but I found the current snippet when I did a search and it works and is more readable.